src/server/req_modify.c

```diff
@@ -387,6 +388,48 @@ req_modifyjob(struct batch_request *preq)
 		return;
 	}
 
+	if(find_sched_from_sock(preq->rq_conn) == NULL)
+	{
+		svrattrl *svrattrl_list;
+		svrattrl *cur_svr;
+		pbs_list_head phead;
+		svrattrl *cur_plist;
+		char *logstr = NULL;
+		int i;
+		CLEAR_HEAD(phead);
+		for (i = 0; i < JOB_ATR_LAST; i++) {
+			if (pjob->ji_wattr[i].at_flags & ATR_VFLAG_MODIFY) {
+				job_attr_def[i].at_encode(&pjob->ji_wattr[i], &phead, job_attr_def[i].at_name, NULL, ATR_ENCODE_CLIENT, &svrattrl_list);
+				for (cur_plist = plist; cur_plist != NULL; cur_plist = (svrattrl *)GET_NEXT(cur_plist->al_link)) {
+					if (strcmp(cur_plist->al_name, job_attr_def[i].at_name) != 0)
+						continue;
+					else {
+						for (cur_svr = svrattrl_list; cur_svr != NULL; cur_svr = (svrattrl *)GET_NEXT(cur_svr->al_link)) {
+							if (pjob->ji_wattr[i].at_type == ATR_TYPE_RESC) {
+								if (cur_plist->al_resc != NULL) {
+									if (strcmp(cur_plist->al_resc, cur_svr->al_resc) == 0) {
+										pbs_asprintf(&logstr, "%s.%s=%s", cur_svr->al_name, cur_svr->al_resc, cur_svr->al_value);
+										break;
+									}
+								}
+							}
+							else {
+								pbs_asprintf(&logstr, "%s=%s", cur_svr->al_name, cur_svr->al_value);
+								break;
+							}
+						}
+					}
+					if (logstr == NULL && cur_plist->al_value[0] == '\0') { /* unset */
+						pbs_asprintf(&logstr, "%s%s%s=UNSET", cur_plist->al_name, cur_plist->al_resc ? "." : "", cur_plist->al_resc ? cur_plist->al_resc : "");
+					}
+					account_record(PBS_ACCT_ALTER, pjob, logstr);
```

Comment:
I can make the design document more explicit, but it does say 'Whenever a job attribute is altered, an 'a' record will be emitted to the accounting log.'  To me this says there will be one 'a' record printed per attribute.

Reply:
I think a more explicit way would be that whenever a job is altered there will be an ‘a’ record logged for each modified attribute.
I do feel this will make logs too verbose but I guess that’s a forum discussion at this point.